component/common/config/types.go

```diff
@@ -0,0 +1,153 @@
+// Package config contains types from github.com/prometheus/common/config, but changes them to be compatible with river
+package config
+
+import (
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// HTTPClientConfig mirrors config.HTTPClientConfig
+type HTTPClientConfig struct {
+	BasicAuth       *BasicAuth     `river:"basic_auth,block,optional"`
+	Authorization   *Authorization `river:"authorization,block,optional"`
+	OAuth2          *OAuth2Config  `river:"oauth2,block,optional"`
+	BearerToken     Secret         `river:"bearer_token,attr,optional"`
+	BearerTokenFile string         `river:"bearer_token_file,attr,optional"`
+	ProxyURL        URL            `river:"proxy_url,attr,optional"`
+	TLSConfig       TLSConfig      `river:"tls_config,block,optional"`
+	FollowRedirects bool           `river:"follow_redirects,attr,optional"`
+	EnableHTTP2     bool           `river:"enable_http2,attr,optional"`
+}
+
+// Convert converts our type to the native prometheus type
+func (h *HTTPClientConfig) Convert() *config.HTTPClientConfig {
+	return &config.HTTPClientConfig{
+		BasicAuth:       h.BasicAuth.Convert(),
+		Authorization:   h.Authorization.Convert(),
+		OAuth2:          h.OAuth2.Convert(),
+		BearerToken:     config.Secret(h.BearerToken),
+		BearerTokenFile: h.BearerTokenFile,
+		ProxyURL:        h.ProxyURL.Convert(),
+		TLSConfig:       *h.TLSConfig.Convert(),
+		FollowRedirects: h.FollowRedirects,
+		EnableHTTP2:     h.EnableHTTP2,
+	}
+}
+
+// DefaultHTTPClientConfig for initializing objects
+var DefaultHTTPClientConfig = HTTPClientConfig{
+	FollowRedirects: true,
+	EnableHTTP2:     true,
+}
+
+// BasicAuth configures Basic HTTP authentication credentials.
+type BasicAuth struct {
+	Username     string `river:"username,attr,optional"`
+	Password     string `river:"password,attr,optional"`
+	PasswordFile string `river:"password_file,attr,optional"`
+}
+
+// Convert converts our type to the native prometheus type
+func (b *BasicAuth) Convert() *config.BasicAuth {
+	if b == nil {
+		return nil
+	}
+	return &config.BasicAuth{
+		Username:     b.Username,
+		Password:     config.Secret(b.Password),
+		PasswordFile: b.PasswordFile,
+	}
+}
+
+// URL mirrors config.URL
+type URL string
+
+// Convert converts our type to the native prometheus type
+func (u URL) Convert() config.URL {
+	if u == "" {
+		return config.URL{}
+	}
+	urlp, _ := url.Parse(string(u))
+
+	return config.URL{URL: urlp}
+}
+
+// Secret mirrors config.Secret
+type Secret string
+
+// Authorization sets up HTTP authorization credentials.
+type Authorization struct {
+	Type            string `river:"type,attr,optional"`
+	Credentials     string `river:"credentials,attr,optional"`
```

Comment:
This needs to use `rivertypes.Secret` instead of `string` to avoid leaking the value at /debug/config.